src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -100,12 +106,17 @@ private void createChildren() {
      */
     @Override
     public String getAttributeValue(String namespace, String localPart) {
-        if (TEXT_ATTRIBUTE_NAME.equals(localPart)) {
-            return ident;
+        final String result;
+        if (VALUE_ATTRIBUTE_NAME.equals(localPart)) {
+            result = value;
+        }
+        else if (TEXT_ATTRIBUTE_NAME.equals(localPart)) {
+            result = ident;
```

Comment:
I am lost a meaning of `private AbstractNode[] attributes;` you place ident or value in this collection but in the same time keep them separate in special fields. What is a reason ?

Reply:
moved all to the map